### DISCUSSION TITLE
Share publicly / Embed dashboard not visible

### DISCUSSION THREAD

**Discussion by greylockglass** [plausible/analytics#1207] [Anything Else]
Hey, all you Plausible people!

So I've toggled my dashboard visibility to public, created a shared link, which I copied below into the Share Link box, generated the embed code, which I dropped into my WordPress site (both in a "Custom HTML" block, and free floating in the code...AND...nothing.  I update the page and nothing shows up on the public facing side.

Turned off Word Fence and CloudFlare to see if that was the problem. No change.
https://www.greylockglass.com/about-the-greylock-glass/

Thoughts?

https://plausible.io/greylockglass.com

Thanks!

the mongrel

WordPress 5.8, BTW

**Comment by metmarkosaric** [comment]
when i look at your source code, i cannot see the correct embed code. this is what i see:

![Screenshot from 2021-07-25 08-02-54](https://user-images.githubusercontent.com/34340819/126889416-ffe01f7e-e22f-4af9-922c-11f7c7bafb0e.png)

this looks different from the code we've given you. could you please make sure you're using the correct embed code that is presented in your Plausible settings. then it should work

**Reply by greylockglass** [reply]
You are correct. That code is from a last-ditch effort to use the generic "Embed" block and URL — obviously that didn't work. 

The reason that useless code is there is that, whether I generate the code and try to drop it in a "Custom HTML" block (recommended in most instances) or insert the iframe directly into the code between blocks, the page will not save the changes.

So...I can't show you what refuses to save.

As I said, I disabled all security to test it to no avail. I suppose I might be overlooking something, but I've been working in WordPress for a decade, and have been building websites since 1998 — I don't tend to make rookie mistakes that often these days...

Further thoughts?

Thanks!

-J.

**Reply by metmarkosaric** [reply]
hmmm difficult to say. I use WordPress myself and embed the dashboard using the Custom HTML block without any issues. It looks like this:

![Screenshot from 2021-07-26 06-42-53](https://user-images.githubusercontent.com/34340819/126934142-d11bfb27-aff7-4c03-b7c4-1a7b515c8641.png)

**Reply by greylockglass** [reply]
Thanks for the reply! Yeah. Same code I'm using, except mine didn't have the` amp;`, which didn't make a difference when I inserted it. I'm flummoxed.

**Reply by greylockglass** [reply]
Okay, so further research reveals that the probable culprit in this situation is the site from which the iframe content is being served, in this case, Plausible. Which explains why the error I get is, "plausible.io refused to connect." See attached screenshot. So whatever the problem is, it's happening on Plausible's end. Does Plausible not allow this function for trial accounts, maybe?


<img width="428" alt="Screen Shot 2021-07-26 at 9 37 23 AM" src="https://user-images.githubusercontent.com/25245603/127002043-60134e14-976c-4ecf-9f53-7eccf088bfd0.png">

**Reply by metmarkosaric** [reply]
That error normally happens if you try to embed a password protected shared link. Did you by any chance password protected the link? It must be open without a password to work as an embed

**Reply by greylockglass** [reply]
I just used the link as provided. I do not see the option to add a password as I read about in the documentation. See attached. I DO see a lock symbol by my name, however. This suggests a decreased level of features...
<img width="821" alt="Screen Shot 2021-07-26 at 10 18 36 AM" src="https://user-images.githubusercontent.com/25245603/127004820-20326c33-29a1-4d1e-b45e-3281c04fd908.png">

**Reply by metmarkosaric** [reply]
That lock means that you created a password protected shared link. You need to edit that link or create a new one without a password if you want to embed the dashboard. You can follow the steps here https://plausible.io/docs/embed-dashboard

We don't have any restrictions on the trial other than the 3p day time limit.

**Reply by greylockglass** [reply]
Update: After a "we're experiencing difficulties" message popped up, I was able to generate an embed code without password protection — via a page/text boxes that had not been presented to me before.

Problem of it not saving still exists, though, and all my research says that WordPress doesn't go for iFrames due to security reasons unless you bypass security with some code in your functions.php file.

- - - - - - - -  - - - - -

Yes, those are the directions I have been following. As I said earlier, "I just used the link as provided. I do not see the option to add a password as I read about in the documentation."

In other words, I actually read in the documentation how to create a password protected link, and that option is not available to me; ergo, if the link is password protected, it's doing it automatically, and I have no way to disable that feature.

**Comment by metmarkosaric** [comment]
Thanks for the update! Hmm very weird. I didn't need to add any code to my functions.php file. It worked like any other embed. We use the same functionality to showcase the stats within the WP dashboard and that works fine too. We haven't had any other reports of this issue.

**Reply by greylockglass** [reply]
Yeah, I don't get it. I DO have other iframes in place here and there, so I can't figure out what might be causing a conflict.

On the other hand, I had no problem once installed the Advanced iFrames plugin. I kind of hate adding yet another plugin to the bloat, but until I can take the time to really dig in to the problem, it will have to do. Thanks for your attention and attempts to troubleshoot with me.

J.

**Reply by metmarkosaric** [reply]
nice! glad to hear you got it working! have fun!
